src/Chainweb/Pact/PactService.hs

```diff
@@ -345,12 +323,7 @@ initializeCoinContract _logger v cid pwo = do
         Nothing -> throwM NoBlockValidatedYet
         (Just !p) -> return p
       let target = Just (succ bhe, bhash)
-      bhdb <- asks _psBlockHeaderDb
-      parentHeader <- liftIO $! lookupM bhdb bhash
-        `catch` \e -> throwM $ InternalPactServiceException
-          { _pactServiceExceptionMsg = "failed lookup of parent header in initializeCoinContract"
-          , _pactServiceExceptionInner = Just e
-          }
+      parentHeader <- lookupBlockHeader bhash "initializeCoinContract"
```

Comment:
Also per https://github.com/kadena-io/chainweb-node/pull/902#discussion_r382709468 

- if the same code block shows up 4 times, please consider making a single function for it

Reply:
That one is addressed here: https://github.com/kadena-io/chainweb-node/pull/935/commits/889fe0e754f5dc21279ff8e9c40b65f57def2013